openreview/api/iThenticate_client.py

```diff
@@ -0,0 +1,233 @@
+            headers=headers,
+            data=file_data,
+        )
+        return response.json()
+
+    def generate_similarity_report(
+        self,
+        submission_id,
+        search_repositories,
+        submission_auto_excludes=None,
+        auto_exclude_self_matching_scope=None,
+        priority=None,
+        view_settings=None,
+        indexing_settings=None,
+    ):
+        data = {"generation_settings": {"search_repositories": search_repositories}}
+        if submission_auto_excludes is not None:
+            data["generation_settings"][
+                "submission_auto_excludes"
+            ] = submission_auto_excludes
+        if auto_exclude_self_matching_scope is not None:
+            data["generation_settings"][
+                "auto_exclude_self_matching_scope"
+            ] = auto_exclude_self_matching_scope
+        if priority is not None:
+            data["generation_settings"]["priority"] = priority
+        if view_settings is not None:
+            data["view_settings"] = view_settings
+        if indexing_settings is not None:
+            data["indexing_settings"] = indexing_settings
+
+        response = requests.put(
+            f"https://{self.TCA_URL}/api/v1/submissions/{submission_id}/similarity",
+            headers=self.headers,
+            json=data,
+        )
+        return response.json()
+
+    def get_similarity_report_status(self, submission_id):
+
+        response = requests.get(
+            f"https://{self.TCA_URL}/api/v1/submissions/{submission_id}/similarity",
+            headers=self.headers,
+        )
+        return response.json()["status"]
+
+    def get_viewer_url(
+        self,
+        submission_id,
+        viewer_id,
+        viewer_default_permission_set,
+        viewer_timestamp,
+        viewer_permissions=None,
+        similarity=None,
+        author_metadata_override=None,
+        sidebar=None,
+    ):
+        data = {
+            "viewer_user_id": viewer_id,
+            "locale": "en-US",
+            "viewer_default_permission_set": viewer_default_permission_set,
+            "eula": {
+                "version": "v1beta",
+                "accepted_timestamp": viewer_timestamp,
+                "language": "en-US",
+            },
+        }
+        if viewer_permissions is not None:
+            data["viewer_permissions"] = viewer_permissions
+        if similarity is not None:
+            data["similarity"] = similarity
+        if viewer_permissions is not None:
+            data["author_metadata_override"] = author_metadata_override
+        if viewer_permissions is not None:
+            data["sidebar"] = sidebar
+        headers = self.headers.copy()
+        headers["Content-Type"] = "application/json"
+        response = requests.get(
+            f"https://{self.TCA_URL}/api/v1/submissions/{submission_id}/viewer-url",
+            headers=headers,
+            json=data,
```

Comment:
do you need to pass a JSON in a GET request?

Reply:
This should be a post request. I've pushed the change.

Reply:
ok, I fixed the conflicts.